AsyncDisplayKit/ASVideoNode.mm

```diff
@@ -128,29 +140,22 @@ - (void)layout
   _spinner.position = CGPointMake(bounds.size.width/2, bounds.size.height/2);
 }
 
-- (void)didLoad
+- (void)setPlaceholderImagefromAsset:(AVAsset*)asset
 {
-  [super didLoad];
-  [[NSNotificationCenter defaultCenter] addObserver:self selector:@selector(didPlayToEnd:) name:AVPlayerItemDidPlayToEndTimeNotification object:nil];
-
-  if (_shouldBePlaying) {
-    _playerNode = [[ASDisplayNode alloc] initWithLayerBlock:^CALayer *{
-      AVPlayerLayer *playerLayer = [[AVPlayerLayer alloc] init];
-      if (!_player) {
-        _player = [AVPlayer playerWithPlayerItem:[[AVPlayerItem alloc] initWithAsset:_asset]];
-        _player.muted = _muted;
-      }
-      playerLayer.player = _player;
-      playerLayer.videoGravity = [self gravity];
-      return playerLayer;
-    }];
+  ASDN::MutexLocker l(_videoLock);
+  if (!_placeholderImageNode)
+    _placeholderImageNode = [[ASImageNode alloc] init];
```

Comment:
It seems to me like this class should just subclass ASImageNode.  There's little cost to do so, and it's still an ASControlNode.